components/nimbus/src/stateful/targeting.rs

```diff
@@ -27,3 +35,71 @@ impl NimbusTargetingHelper {
         }
     }
 }
+
+pub trait RecordedContext: Send + Sync {
+    /// Returns a JSON representation of the context object
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn to_json(&self) -> JsonObject;
+
+    /// Returns a HashMap representation of the event queries that will be used in the targeting
+    /// context
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn get_event_queries(&self) -> JsonObject;
+
+    /// Sets the object's internal value for the event query values
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn set_event_query_values(&self, json: JsonObject);
+
+    /// Records the context object to Glean
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn record(&self);
+}
+
+impl dyn RecordedContext {
+    pub fn execute_queries(
+        &self,
+        nimbus_targeting_helper: &NimbusTargetingHelper,
+    ) -> Result<Map<String, Value>> {
+        let results = Map::from_iter(self.get_event_queries().iter().filter_map(
+            |(key, value): (&String, &Value)| match value.as_str() {
+                Some(v) => match EventQueryType::validate_query(v) {
```

Comment:
I was under the impression we were only going to do this for debug builds and that we were going to cause a hard error e.g., crash, so that invalid queries couldn't land?